### DISCUSSION TITLE
What does CANCELED mean amongst build steps and should I worry?

### DISCUSSION THREAD

**Discussion by jeffrson** [docker/buildx#1157] [Q&A]
Hi,
I'm trying to build images for two platforms: linux/amd64 and linux/arm64
Now when I run `docker buildx build` sometimes I get CANCELED for one of those platforms, sometimes for the other:
```
#9 [linux/arm64 builder 2/5] WORKDIR /usr/src/app
#9 CANCELED
```

```
#11 [linux/amd64 builder 2/5] WORKDIR /usr/src/app
#11 CANCELED
```

Should I'll be worried about this?

**Comment by tonistiigi** [comment]
> Should I'll be worried about this?

No, but if you have a reproducer you can post it and we can take a look what condition causes it. This can happen when buildkit runs two steps in parallel but then realizes after some event that one of them is not needed anymore because of a cache match further ahead. Then it cancels the other step. Normally we have protections for detecting such cases so you wouldn't see this often.

We have been seeing some new reports of this after introducing mergeop so maybe you are using it as well? But nobody has provided a reproducer afaik.

**Reply by bvobart** [reply]
> This can happen when buildkit runs two steps in parallel but then realizes after some event that one of them is not needed anymore because of a cache match further ahead. Then it cancels the other step.

Is there any way to disable that cancellation? 

For context, I have a Dockerfile with four targets: `build`, `lint`, `test`, `release`. They all depend on `build` but have no further dependencies on one other. Obviously, if there is an error in `build`, then `lint`, `test` and `release` cannot continue. However, I'm noticing that a failure in the build for `lint`, results in `test` and `release` being cancelled and vice versa. Is there some kind of way to make the builder continue running `lint`, `test` and `release` all to completion, regardless of whether any of those fail? That would allow me to see linting errors, as well as test errors, all in one build output.

<sub>in case it matters, I'm using `docker buildx bake` to build to to these targets, though all targets are in the same Dockerfile.</sub>

**Comment by jeffrson** [comment]
Thanks you for those hints and information!

I'll see if I could find a repro.
